plugins/module_utils/crypto/cryptography_support.py

```diff
@@ -68,24 +69,53 @@ def cryptography_get_extensions_from_cert(cert):
     # Since cryptography won't give us the DER value for an extension
     # (that is only stored for unrecognized extensions), we have to re-do
     # the extension parsing outselves.
+    backend = default_backend()
+    try:
+        # For certain old versions of cryptography, backend is a MultiBackend object,
+        # which has no _lib attribute. In that case, revert to the old approach.
+        x = backend._lib
+    except AttributeError:
+        backend = cert._backend
+
     result = dict()
-    backend = cert._backend
-    x509_obj = cert._x509
-
-    for i in range(backend._lib.X509_get_ext_count(x509_obj)):
-        ext = backend._lib.X509_get_ext(x509_obj, i)
-        if ext == backend._ffi.NULL:
-            continue
-        crit = backend._lib.X509_EXTENSION_get_critical(ext)
-        data = backend._lib.X509_EXTENSION_get_data(ext)
-        backend.openssl_assert(data != backend._ffi.NULL)
-        der = backend._ffi.buffer(data.data, data.length)[:]
-        entry = dict(
-            critical=(crit == 1),
-            value=base64.b64encode(der),
-        )
-        oid = obj2txt(backend._lib, backend._ffi, backend._lib.X509_EXTENSION_get_object(ext))
-        result[oid] = entry
+    try:
+        x509_obj = cert._x509
+
+        for i in range(backend._lib.X509_get_ext_count(x509_obj)):
```

Comment:
It is a code smell to wrap big blocks of code with try-except. If anything unexpected on any line raises `AttributeError`, expect surprises in the control flow. You could probably put this in the else-block, though.

Reply:
I rewrote it to minimize the amount of lines covered by `try`. This is somewhat less efficient, since the decision which code-path is taken is now done once per iteration, but it definitely improves readability. I hope that sooner or later we can get rid of these functions anyway, resp. rewrite them so that the happy path doesn't require any internals.

Changed in 548cdc698579d9d47d44bef626eeda47b05da039.